demos/text_detection_demo/cpp/src/cnn.cpp

```diff
@@ -119,3 +102,185 @@ InferenceEngine::BlobMap Cnn::Infer(const cv::Mat &frame) {
 
     return blobs;
 }
+
+std::string EncoderDecoderCNN::check_net_names(std::vector<std::string> output_names_encoder,
+                                               std::vector<std::string> input_names_decoder,
+                                               std::vector<std::string> output_names_decoder) {
+    if (std::find(
+            output_names_encoder.begin(),
+            output_names_encoder.end(),
+            out_enc_hidden_name_) == output_names_encoder.end())
+        return out_enc_hidden_name_;
+    if (std::find(output_names_encoder.begin(),
+            output_names_encoder.end(),
+            features_name_) == output_names_encoder.end())
+        return features_name_;
+    if (std::find(input_names_decoder.begin(),
+            input_names_decoder.end(),
+            in_dec_hidden_name_) == input_names_decoder.end())
+        return in_dec_hidden_name_;
+    if (std::find(input_names_decoder.begin(),
+            input_names_decoder.end(),
+            features_name_) == input_names_decoder.end())
+        return features_name_;
+    if (std::find(input_names_decoder.begin(),
+            input_names_decoder.end(),
+            in_dec_symbol_name_) == input_names_decoder.end())
+        return in_dec_symbol_name_;
+    if (std::find(output_names_decoder.begin(),
+            output_names_decoder.end(),
+            out_dec_hidden_name_) == output_names_decoder.end())
+        return out_dec_hidden_name_;
+    if (std::find(output_names_decoder.begin(),
+            output_names_decoder.end(),
+            out_dec_symbol_name_) == output_names_decoder.end())
+        return out_dec_symbol_name_;
+    return std::string("");
+ }
+
+void EncoderDecoderCNN::Init(const std::string &model_path, Core & ie, const std::string & deviceName, const cv::Size &new_input_resolution) {
+    // ---------------------------------------------------------------------------------------------------
+    // --------------------------- Checking paths --------------------------------------------------------
+    std::string model_path_decoder = model_path;
+    auto network_encoder = ie.ReadNetwork(model_path);
+    CNNNetwork network_decoder;
+    try {
+        if (model_path_decoder.find("encoder") == std::string::npos)
+            throw std::runtime_error("Model path does not contain 'encoder'");
```

Comment:
You should create your own type to throw. This will remove a need to parse error message outside

Reply:
Done